models/media_stats.sql

```diff
@@ -0,0 +1,126 @@
+{{ 
+  config(
+    materialized='table',
+  )
+}}
+
+with prep as (
+
+ 	select 
+    p.media_id,
+    p.duration,
+    p.title,
+    p.media_type,
+    p.media_player_type,
+    sum(p.play_time_sec) / 3600 as play_time_h,
+    avg(p.play_time_sec / 60) as avg_play_time_min,
+    sum(case when p.is_complete_play then 1 else 0 end) as complete_plays,
+    count(distinct p.page_view_id) as views,
+    count(distinct p.domain_sessionid) as "sessions",
+    avg(coalesce(p.play_time_sec / nullif(p.duration, 0), 0)) as retention_rate,
+    count(distinct p.domain_userid) as unique_users
+
+  from {{ ref("plays_by_pageview") }} p
+ 
+  group by 1, 2, 3, 4, 5
+
+)
+
+, valid_play_stats as (
+
+    select
+      media_id,
+      min(start_tstamp) as first_play,
+      max(start_tstamp) as last_play,
+      count(*) as valid_plays
+    
+    from {{ ref("plays_by_pageview") }}
+
+    where is_valid_play
+
+    group by 1
+
+)
+
+, impressions as (
+
+    select 
+      media_id,
+      count(*) as impressions
+    
+    from {{ ref("interactions") }}
+
+    where event_type = 'ready'
+
+    group by 1
+
+)
+
+, percent_passed as (
+
+    select
+      media_id, 
+      {% for key, value in var('snowplow__percent_progress_boundaries').items() %}
+        sum(case when _{{ key }}_percent_passed > 0 then 1 else 0 end) as _{{ key }}_percent_passed
+        {% if not loop.last %}
+          ,
+        {% endif %}
+      {% endfor %}
+
+    from {{ ref("plays_by_pageview") }}
+
+    group by 1
+)
+
+, user_stats as (
```

Comment:
Not sure how much these add. More generally I am not sure we need `plays_by_user`. It feels like a slightly odd level of aggregation, essentially a complete view of a single video per user. What do you think?